### ISSUE TITLE
Build s3fs with WinFSP

### ISSUE THREAD

**Issue by ggtakec** [s3fs-fuse/s3fs-fuse#728]
## Details about issue
About trials to be able to build and execute with WinFSP/Cygwin.
The beginning was PR #712 from @chrilith, and wrote comments by @billziss-gh who is it's author and @feoff3.
I thought it would be better to record it as this Issue of s3fs rather than continuing to follow Issue with PR comments.

- **Google group**  
https://groups.google.com/forum/#!topic/winfsp/xXe6M_MoW8w
- **WinFSP**  
https://github.com/billziss-gh/winfsp
- **First Trigger**  
#712 (please see this PR)
- **Current Issue** on WinFSP  
https://github.com/billziss-gh/winfsp/issues/143

We will actively participate if we can help, and we are happy to help s3fs work on windows.

**Comment by mosi-kha** [comment]
any activity on this issue?

**Comment by gaul** [comment]
Windows support is an important feature that we would gladly accept pull requests for.  Unfortunately the current s3fs team lacks the required expertise.

**Comment by ggtakec** [comment]
I forgot this issue for a while. I'm sorry.

We have to think about using WinFSP, but I learned another approach from sshfs.

And I have confirmed that s3fs works on current Windows now.
It's easy to do with a little Linux knowledge.

With the combination of WSL2 + s3fs that can be used on Windows 10/11, I was able to confirm access to the S3 bucket as a Windows Network Drive.
I confirmed it as follows.

- Enable WSL2 on Windows 10(11)
- Start Ubuntu 20.04 on WSL2
- Prepare files on Ubuntu 20.04  
Modify /etc/fuse.conf(set user_allow_other) and create /etc/passwd-s3fs.  
- Start s3fs as root.  
Similar to booting s3fs on Linux.  
You must run s3fs as root.  
For example, mount the bucket on /mnt/s3.
- On the Windows side, set `\\wsl$\Ubuntu-20.04` as Network Drive (as Z: drive).  
It can be set from Explorer etc.

With the above operation, I could check the S3 bucket from Windows(Explorer) under `Z:\mnt\s3`.
